src/main/java/com/projects/quickcart/controller/ProductController.java

```diff
@@ -17,6 +18,23 @@ public ModelAndView productsView() {
 		ModelAndView mView = new ModelAndView("products");
 		mView.addObject("producs", bService.getProducts());
 		return mView;
+
 	}
 
+	@GetMapping("/{productId}")
+	public ModelAndView getProductsView(@PathVariable("productId") int productId) {
+		ModelAndView mView = new ModelAndView("products");
+		mView.addObject("products", bService.getProduct());
+
+		return mView;
+
+	}
+
+	@GetMapping("/{category}")
```

Comment:
category is not the path variable it should be a **query parameter** and the path should be like

`/products?category={category}`